storage/diskmetricstore.go

```diff
@@ -345,3 +453,110 @@ func extractPredefinedHelpStrings(g prometheus.Gatherer) (map[string]string, err
 	}
 	return result, nil
 }
+
+func newPushTimestampGauge(groupingLabels map[string]string, t time.Time) *dto.MetricFamily {
+	return newTimestampGauge(pushMetricName, pushMetricHelp, groupingLabels, t)
+}
+
+func newPushFailedTimestampGauge(groupingLabels map[string]string, t time.Time) *dto.MetricFamily {
+	return newTimestampGauge(pushFailedMetricName, pushFailedMetricHelp, groupingLabels, t)
+}
+
+func newTimestampGauge(name, help string, groupingLabels map[string]string, t time.Time) *dto.MetricFamily {
+	var ts float64
+	if !t.IsZero() {
```

Comment:
Why do you need this if?

Reply:
Because the zero value of time.Time is not unixtime 0. Unixtime zero is 1970-01-01 00:00:00 UTC, Go time.Time zero value is 0001-01-01 00:00 UTC).